msal-dotnet-articles/acquiring-tokens/desktop-mobile/linux-broker-net.md

```diff
@@ -13,77 +13,137 @@ ms.topic:    how-to
     }
+}
 ```
 
 ## Sample App 
-There is a sample app in MSAL.NET repo, it's under ./tests/devapps/WAM/NetWSLWam path. The sample app has a dependenc of `libx11-dev` package. Please run `apt install libx11-dev` to install the package. To run the sample app, just run the command below.
+A sample application is available for developers who want to try the authentication broker on Linux. The application is located [in the MSAL.NET GitHub repository](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/tree/main/tests/devapps/WAM/NetWSLWam). The app has a dependency of `libx11-dev` package. 
+
+On Debian-based distributions, run `sudo apt install libx11-dev` to install the package. The `libx11` library is used to get the console window handle on Linux. Here's the sample code to use `libx11` to get the window handle:
+
+```csharp
+using System;
+using System.Runtime.InteropServices;
+
+class X11Interop
+{
+    [DllImport("libX11")]
+    public static extern IntPtr XOpenDisplay(IntPtr display);
+
+    [DllImport("libX11")]
+    public static extern IntPtr XDefaultRootWindow(IntPtr display);
+
+    public static void Main()
+    {
+        IntPtr display = XOpenDisplay(IntPtr.Zero);
+        if (display == IntPtr.Zero)
+        {
+            Console.WriteLine("Unable to open X display.");
+            return;
+        }
+
+        IntPtr rootWindow = XDefaultRootWindow(display);
+        Console.WriteLine($"Root window handle: {rootWindow}");
+    }
+}
 ```
+
+We recommend play with the sample app using your client id. To run the sample app, use the following command:
+```dotnetcli
 dotnet run --project tests\devapps\WAM\NetWSLWam\test.csproj
 ```
 
 ## WSL Scenario
 
 ### Update to the latest version of WSL
-Please make sure you have updated to the most recent version of WSL. WAM Account Control dialog is supported from WSL version 2.4.13. Otherwise, the broker scenario won't work. Here is the command to update WSL.
+
+Enure you have updated to the latest WSL release. The WAM Account Control dialog is supported in WSL versions 2.4.13 and above. Using the broker isn't possible with earlier versions. To check the WSL version, use the following command:
+
+```powershell
+wsl --version
 ```
-wsl.exe --update
+
+To update WSL, run the following command from Windows Terminal:
+
+```powershell
+wsl --update
 ```
 
 ### Set up Keyring in WSL
-We use libsecret on Linux and it needs to communicate with keyring daemon. Users can use `seahorse`(gnome GUI) package to manage the keyring with a user interface. You need to first install the package by `apt install seahorse` and then follow the instructions below.
+MSAL uses `libsecret` on Linux. It is required to communicate with the `keyring` daemon. Users can use [Seahorse](https://wiki.gnome.org/Apps/Seahorse/) (a GNOME application for managing encryption keys and passwords) to manage the `keyring` contents through a Graphical User Interface (GUI).
```

Comment:
It could be helpful to add more details on how Keyring plays a role with the library and WAM. Keyring might be confused with WAM in some way.